ml-dsa/src/lib.rs

```diff
@@ -668,6 +678,16 @@ impl<P: MlDsaParams> VerifyingKey<P> {
         }
     }
 
+    /// Deterministically generate a verifying key from the specified seed.
+    ///
+    /// This method reflects the ML-DSA.KeyGen_internal algorithm from FIPS 204, but only returns a
+    /// verifying key.
+    #[must_use]
+    pub fn from_seed(seed: &B32) -> Self {
+        let kp = P::from_seed(seed);
+        kp.verifying_key
+    }
```

Comment:
I'm a little bit wary about an API like this on `VerifyingKey` because it might lead a user to confuse their seed with a public key.

How about putting `from_seed` on `KeyPair`, and then the user can call `verifying_key()` instead? It also avoids the duplication of having two `from_seed` methods.